src/CarbonAware.WebApi/test/unitTests/HttpResponseExceptionFilterTests.cs

```diff
@@ -136,12 +137,79 @@ public void TestOnException_GenericException()
         Assert.AreEqual("Exception", content.Title);
         Assert.AreEqual("My validation error", content.Detail);
     }
+
+    [Test]
+    public void TestOnException_GenericException_WithVerboseTrue()
+    {
+        var inMemorySettings = new Dictionary<string, string> {
+            { $"{CarbonAwareVariablesConfiguration.Key}:VerboseApi", "true" }
+        };
+        this._config = new ConfigurationBuilder()
+            .AddInMemoryCollection(inMemorySettings)
+            .Build();
+
+        // Arrange
+        var ex = new Exception("My validation error");
+        var exceptionContext = new ExceptionContext(this._actionContext, new List<IFilterMetadata>())
+        {
+            Exception = ex
+        };
+        
+        var filter = new HttpResponseExceptionFilter(this._logger.Object, this._config);
+
+        // Act
+        filter.OnException(exceptionContext);
+
+        var result = exceptionContext.Result as ObjectResult ?? throw new Exception();
```

Comment:
Seems weird to throw in exception in a test like this, aren't we guaranteed to know the object parses? Maybe do `exceptionContext.Result as ObjectResult!` or something of the like? If not maybe pass a response message to the exception saying it should be a null case. Just seems like it'd be a vague error to look into if it fails

Reply:
Good point @JenMadiedo . CC'ing @pritipath

Reply:
Fixed.